source/SH3/graphics/texture.cpp

```diff
@@ -4,44 +4,57 @@
  *
  *  @copyright 2016  Palm Studios
  *
- *
  *  @date 2-1-2017
  *
  *  @author Jesse Buhagiar
  */
 #include <SH3/graphics/texture.hpp>
 #include <SH3/system/log.hpp>
+#include <SH3/arc/vfile.hpp>
+
 
 using namespace sh3_graphics;
 
-std::string texture::load_error::message() const
+//TODO: Scale the texture and then
+int sh3_texture::Load(sh3_arc& mft, const std::string& filename)
 {
-    std::string error;
-    switch(result)
-    {
-    case load_result::SUCCESS:
-        error = "Success";
-        break;
-    case load_result::INVALID_CHUNK:
-        error = "Invalid chunk";
-        break;
-    case load_result::END_OF_FILE:
-        error = "End of file";
-        break;
-    }
-    return error;
-};
-
-texture::load_error sh3_graphics::texture::Load(const std::string& filename, sh3_arc& arc)
-{
-    load_error error;
+    sh3_texture_header header;
+    sh3_arc_vfile file(mft, filename);
+    sh3_arc_vfile::read_error e;
+
+    int ret = file.ReadData(&header, sizeof(sh3_texture_header), e); // Read in the header
```

Comment:
Should be `std::size_t`.
I"m also in favor of using `sizeof(header)` (generally `sizeof(variable)` where applicable) because the intention here is to read the size of the variable `header`, which just happens to be a `sh3_texture_header` here.